lib/workers/repository/update/branch/auto-replace.ts

```diff
@@ -198,11 +199,15 @@ export async function doAutoReplace(
           newName
         );
       }
-      if (currentDigest && newDigest) {
-        newString = newString.replace(
-          regEx(escapeRegExp(currentDigest), 'g'),
-          newDigest
-        );
+      if (newDigest) {
+        if (currentDigest) {
+          newString = newString.replace(
+            regEx(escapeRegExp(currentDigest), 'g'),
+            newDigest
+          );
+        } else if (pinDigests) {
+          newString = `${newString}@${newDigest}`;
```

Comment:
this only works for docker images and will break all other managers with different dep types.

Reply:
@viceice - good feedback.  

do you have a suggestion on how to best proceed... would an additional guard make sense to check if it's a docker update?  

alternatively, perhaps `confirmIfDepUpdated` function could be updated to better handle this edge-case

Reply:
I've pivoted to a different solution to handle this edge-case.  explanation added as a comment below